File: eco_restoration/cyboquatic_daily/20260816/lua/canal_blast_radius.lua
local function clamp01(value)
    if value < 0.0 then
        return 0.0
    end
    if value > 1.0 then
        return 1.0
    end
    return value
end

local function number_argument(arguments, index, name)
    local value = tonumber(arguments[index])
    assert(value, name .. " must be numeric")
    return value
end

local function assess(breach_flow_lps, surcharge_duration_s, bank_sensitivity, distance_m, energyreqJ, delta_vt)
    assert(breach_flow_lps > 0.0, "breach_flow_lps must be positive")
    assert(surcharge_duration_s > 0.0, "surcharge_duration_s must be positive")
    assert(bank_sensitivity >= 0.0 and bank_sensitivity <= 1.0, "bank_sensitivity must be 0..1")
    assert(distance_m >= 0.0 and energyreqJ >= 0.0 and delta_vt >= 0.0, "remaining inputs must be non-negative")

    local base_radius_m = math.sqrt(breach_flow_lps * surcharge_duration_s) / 10.0
    local conservative_radius_m = base_radius_m * (1.0 + bank_sensitivity * 1.5)
    local exposure = 0.0
    if conservative_radius_m > 0.0 then
        exposure = clamp01(1.0 - distance_m / conservative_radius_m)
    end

    local energy_load = clamp01(energyreqJ / 1000000.0)
    local velocity_load = clamp01(delta_vt / 10.0)
    local harm_risk = clamp01(
        0.60 * exposure + 0.20 * bank_sensitivity + 0.10 * energy_load + 0.10 * velocity_load
    )
    local knowledge_factor = clamp01(1.0 - 0.35 * bank_sensitivity - 0.25 * energy_load)
    local eco_impact_value = clamp01((1.0 - harm_risk) * (0.40 + 0.60 * knowledge_factor))

    local zone, machine_action = "SAFE", "OPERATE_LOW_IMPACT"
    if harm_risk >= 0.60 then
        zone, machine_action = "EXCLUDE", "NO_ENTRY"
    elseif harm_risk > 0.25 then
        zone, machine_action = "CAUTION", "HOLD_FOR_INSPECTION"
    end

    return {
        base_radius_m = base_radius_m,
        conservative_radius_m = conservative_radius_m,
        knowledge_factor = knowledge_factor,
        eco_impact_value = eco_impact_value,
        harm_risk = harm_risk,
        zone = zone,
        machine_action = machine_action
    }
end

if #arg ~= 6 then
    io.stderr:write(
        "usage: lua canal_blast_radius.lua <breach_flow_lps> <surcharge_duration_s> " ..
        "<bank_sensitivity_0_to_1> <distance_m> <energyreqJ> <delta_vt>\n"
    )
    os.exit(64)
end

local ok, result = pcall(
    assess,
    number_argument(arg, 1, "breach_flow_lps"),
    number_argument(arg, 2, "surcharge_duration_s"),
    number_argument(arg, 3, "bank_sensitivity"),
    number_argument(arg, 4, "distance_m"),
    number_argument(arg, 5, "energyreqJ"),
    number_argument(arg, 6, "delta_vt")
)

if not ok then
    io.stderr:write("error: " .. result .. "\n")
    os.exit(65)
end

print(string.format("base_radius_m=%.3f", result.base_radius_m))
print(string.format("conservative_radius_m=%.3f", result.conservative_radius_m))
print(string.format("knowledge_factor=%.3f", result.knowledge_factor))
print(string.format("eco_impact_value=%.3f", result.eco_impact_value))
print(string.format("harm_risk=%.3f", result.harm_risk))
print("zone=" .. result.zone)
print("machine_action=" .. result.machine_action)

File: eco_restoration/cyboquatic_daily/20260816/cpp/canal_blast_radius.cpp
#include <algorithm>
#include <cmath>
#include <cstdlib>
#include <iomanip>
#include <iostream>
#include <stdexcept>
#include <string>

struct Assessment {
    double base_radius_m;
    double conservative_radius_m;
    double knowledge_factor;
    double eco_impact_value;
    double harm_risk;
    std::string zone;
    std::string machine_action;
};

static double clamp01(double value) {
    return std::clamp(value, 0.0, 1.0);
}

static Assessment assess(
    double breach_flow_lps,
    double surcharge_duration_s,
    double bank_sensitivity,
    double distance_m,
    double energyreqJ,
    double delta_vt
) {
    if (breach_flow_lps <= 0.0 || surcharge_duration_s <= 0.0 || distance_m < 0.0 ||
        energyreqJ < 0.0 || delta_vt < 0.0 || bank_sensitivity < 0.0 || bank_sensitivity > 1.0) {
        throw std::invalid_argument("inputs must be non-negative; flow and duration must be positive; sensitivity must be 0..1");
    }

    const double base_radius_m = std::sqrt(breach_flow_lps * surcharge_duration_s) / 10.0;
    const double conservative_radius_m = base_radius_m * (1.0 + bank_sensitivity * 1.5);
    const double exposure = conservative_radius_m <= 0.0 ? 0.0 : clamp01(1.0 - distance_m / conservative_radius_m);
    const double energy_load = clamp01(energyreqJ / 1000000.0);
    const double velocity_load = clamp01(delta_vt / 10.0);
    const double harm_risk = clamp01(0.60 * exposure + 0.20 * bank_sensitivity + 0.10 * energy_load + 0.10 * velocity_load);
    const double knowledge_factor = clamp01(1.0 - 0.35 * bank_sensitivity - 0.25 * energy_load);
    const double eco_impact_value = clamp01((1.0 - harm_risk) * (0.40 + 0.60 * knowledge_factor));

    std::string zone = "SAFE";
    std::string action = "OPERATE_LOW_IMPACT";
    if (harm_risk >= 0.60) {
        zone = "EXCLUDE";
        action = "NO_ENTRY";
    } else if (harm_risk > 0.25) {
        zone = "CAUTION";
        action = "HOLD_FOR_INSPECTION";
    }

    return {base_radius_m, conservative_radius_m, knowledge_factor, eco_impact_value, harm_risk, zone, action};
}

int main(int argc, char** argv) {
    if (argc != 7) {
        std::cerr << "usage: " << argv[0]
                  << " <breach_flow_lps> <surcharge_duration_s> <bank_sensitivity_0_to_1>"
                  << " <distance_m> <energyreqJ> <delta_vt>\n";
        return 64;
    }

    try {
        const Assessment result = assess(
            std::stod(argv[1]), std::stod(argv[2]), std::stod(argv[3]),
            std::stod(argv[4]), std::stod(argv[5]), std::stod(argv[6])
        );

        std::cout << std::fixed << std::setprecision(3)
                  << "base_radius_m=" << result.base_radius_m << '\n'
                  << "conservative_radius_m=" << result.conservative_radius_m << '\n'
                  << "knowledge_factor=" << result.knowledge_factor << '\n'
                  << "eco_impact_value=" << result.eco_impact_value << '\n'
                  << "harm_risk=" << result.harm_risk << '\n'
                  << "zone=" << result.zone << '\n'
                  << "machine_action=" << result.machine_action << '\n';
        return 0;
    } catch (const std::exception& error) {
        std::cerr << "error: " << error.what() << '\n';
        return 65;
    }
}
